src/async.ts

```diff
@@ -354,17 +360,13 @@ export class AsyncMutex {
    * one who had actually acquired the lock.
    */
   public release(): void {
-    if (this.held) {
+    if (this.held !== 0) {
       if (this.awaiters.length) {
-        setTimeout(() => {
```

Comment:
This `setTimeout` wasn't useful— it was a side effect of my brain being stuck in the past where a callback might really _do_ things, like if we called it directly we could find ourselves executing in a code path that's part of the implementation of a different test, which could have unintended effects based on how test scopes are managed. But that can't happen when the callback is just resolving a promise; continuation handlers for promises are always deferred.